src/plugins/intel_gpu/src/graph/graph_optimizer/fuse_constant_transposes.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "pass_manager.h"
+#include "program_helpers.h"
+
+#include "permute_inst.h"
+#include "fully_connected_inst.h"
+
+using namespace cldnn;
+
+format fuse_constant_transposes::convert_weights_format_by_order(format fmt, const std::vector<uint16_t>& order) const {
+    format weights_fmt = from_weights_layout(to_weights_layout(fmt, false));
+    std::string old_order = weights_fmt.order();
+    std::string new_order = old_order;
+
+    for (size_t i = 0; i < order.size(); ++i) {
+        new_order[i] = old_order[order[i]];
+    }
+
+    return format::find_format(new_order, fmt.block_sizes());
+}
+
+void fuse_constant_transposes::run(program& p) {
+    std::function<bool(const cldnn::program_node*)> is_matmul_weights_path =
+        [&is_matmul_weights_path](const cldnn::program_node* node) {
+        if (node->get_users().empty())
+            return false;
+
+        const auto* next_node = node->get_users().front();
+
+        if (next_node->is_type<fully_connected>()) {
+            size_t weights_offset = next_node->get_primitive()->input_size();
+            return &next_node->get_dependency(weights_offset) == node;
+        }
+
+        if (node->is_constant() && node->get_users().size() == 1)
+            return is_matmul_weights_path(next_node);
+
+        return false;
+    };
+
+    auto itr = p.get_processing_order().begin();
+    while (itr != p.get_processing_order().end()) {
+        auto& node = *itr++;
+
+        if (!node->is_type<permute>())
+            continue;
+
+        auto& permute_node = node->as<permute>();
+
+        if (!is_matmul_weights_path(&permute_node) || !permute_node.get_dependency(0).is_type<data>())
```

Comment:
Why is it limited to weights path only? I suppose it may be useful for other cases as well. For example, after https://github.com/openvinotoolkit/openvino/pull/19516 same transform can be applied to deconvolution weights